internal/util/util.go

```diff
@@ -311,3 +311,11 @@ func Clone[T any](t *T) *T {
 	ret := *t
 	return &ret
 }
+
+// Greatest Common Divisor (Euclidean Algorithm)
+func Gcd[N constraints.Integer](a, b N) N {
```

Comment:
i dont see where this is used

Reply:
By the engine to calculate the needed wait duration.

Reply:
Removed